lib/websocket/websocket-handler.ts

```diff
@@ -0,0 +1,238 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import _ = require('lodash');
+import Promise = require('bluebird');
+import bunyan = require('bunyan');
+import uuid = require('node-uuid');
+import SocketIO = require('socket.io');
+import webSocket = require('ws');
+import Subscription = require('../subscription/subscription');
+import SubscriptionStore = require('../subscription/subscription-store');
+import conf = require('../config');
+import Interface = require('../interface');
+import blpSession = require('../middleware/blp-session');
+import SocketIOWrapper = require('./socket-io-wrapper');
+import WSWrapper = require('./ws-wrapper');
+
+// GLOBAL
+var LOGGER: bunyan.Logger = bunyan.createLogger(conf.get('loggerOptions'));
+
+// PUBLIC FUNCTIONS
+export function sioOnConnect(s: SocketIO.Socket): void
+{
+    var socket: Interface.ISocket = new SocketIOWrapper(s);
+    onConnect(socket);
+}
+
+export function wsOnConnect(s: webSocket): void
+{
+    var socket: Interface.ISocket = new WSWrapper(s);
+    onConnect(socket);
+}
+
+// PRIVATE FUNCTIONS
+function onConnect(socket: Interface.ISocket): void
+{
+    initialize(socket)
+        .then(setup)
+        .catch((err: Error): any => {
+            socket.log.error(err);
+            if (socket.isConnected()) {
+                socket.emit('err', { message: 'Unexpected error: ' + err.message });
+                socket.disconnect();
+            }
+        });
+}
+
+function initialize(socket: Interface.ISocket): Promise<Interface.ISocket>
+{
+    // Setup logging for socket
+    socket.log = LOGGER.child({req_id: uuid.v4()});
```

Comment:
To make the `ISocket` abstraction cleaner, I would recommend that callers call a function to setup logging instead of directly modifying/setting the `log` property.